File: ~Programming_cheat_sheets/_scripting_languages/Python/data-structures-2020/__sorting/selection-sort.py
#------------------------------ Selection sort

def selectionSort(arr):
    for i in range(len(arr)-1):
        minIdx = i
        for j in range(i, len(arr)):
            if arr[j] < arr[minIdx]:
                minIdx = j
        arr[i], arr[minIdx] = arr[minIdx], arr[i]

if __name__ == "__main__":
    arr = [5,4,6,3,7,2,8,1,0,9]
    selectionSort(arr)
    print(arr)
    # [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]

File: ~other/~NEEDS_REVIEW/_compiled_languages/Java/_data-structures-and-algorithms/__sorting-algorithms/SelectionSort.java
package com.example;

import java.util.Arrays;

class SelectionSort {
  public static void sort(int[] arr){
      for(int i=0; i<arr.length-1;i++){ // we don't do it for the last item
          int minIndex = i;

          // get min
          for (int j=i; j<arr.length;j++){
              if (arr[j] < arr[minIndex]){
                  minIndex = j;
              }
          }
          swap(arr, i, minIndex);
      }
  }

  private static void swap(int[] arr, int a, int b){
      int temp = arr[a];
      arr[a] = arr[b];
      arr[b] = temp;
  }
}

public class SelectionSortExample {
    public static void main(String[] args) {
        int[] arr = {
                5,6,4,7,3,8,2,9,1,0
        };

        SelectionSort.sort(arr);

        System.out.println(Arrays.toString(arr));
        //=> [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
    }
}
